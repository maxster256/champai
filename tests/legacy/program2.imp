DECLARE
    sito(2:100);
    a;
    b;
    c;
IN
    sito(4) := 7;
    sito(3) := 60 % sito(4);
    a := sito(4) % sito(3);
    WRITE sito(3);
END
