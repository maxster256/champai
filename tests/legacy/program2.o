SUB A A # BEGIN_CONST_GEN
INC A
INC A
INC A
SUB H H # BEGIN_CONST_GEN
INC H
INC H
INC H
INC H
INC H
INC H
INC H
STORE H
SUB B B # BEGIN_CONST_GEN
INC B
ADD B B
INC B
ADD B B
INC B
ADD B B
INC B
ADD B B
ADD B B
SUB A A # BEGIN_CONST_GEN
INC A
INC A
INC A
LOAD H # GETVAL
SUB G G # BEGIN_CONST_GEN
INC G
INC G
SUB H G
SUB G G # BEGIN_CONST_GEN
INC G
ADD H G
COPY A H
LOAD C
SUB D D # BEGIN MOD/DIV
SUB E E
INC E
COPY F B
JZERO C 68
COPY H C
INC H
SUB H B
JZERO H 47
JUMP 50
ADD C C
ADD E E
JUMP 42
COPY H C
SUB H F
JZERO H 54
JUMP 56
SUB F C
ADD D E
HALF C
HALF E
JZERO E 69
COPY H C
SUB H F
JZERO H 63
JUMP 65
SUB F C
ADD D E
HALF C
HALF E
JUMP 58
SUB F F
SUB A A # BEGIN_CONST_GEN
INC A
INC A
STORE F # STORE_VAL
SUB A A # BEGIN_CONST_GEN
INC A
INC A
INC A
LOAD H # GETVAL
SUB G G # BEGIN_CONST_GEN
INC G
INC G
SUB H G
SUB G G # BEGIN_CONST_GEN
INC G
ADD H G
COPY A H
LOAD B
SUB A A # BEGIN_CONST_GEN
INC A
INC A
LOAD H # GETVAL
SUB G G # BEGIN_CONST_GEN
INC G
INC G
SUB H G
SUB G G # BEGIN_CONST_GEN
INC G
ADD H G
COPY A H
LOAD C
SUB D D # BEGIN MOD/DIV
SUB E E
INC E
COPY F B
JZERO C 131
COPY H C
INC H
SUB H B
JZERO H 110
JUMP 113
ADD C C
ADD E E
JUMP 105
COPY H C
SUB H F
JZERO H 117
JUMP 119
SUB F C
ADD D E
HALF C
HALF E
JZERO E 132
COPY H C
SUB H F
JZERO H 126
JUMP 128
SUB F C
ADD D E
HALF C
HALF E
JUMP 121
SUB F F
SUB A A # BEGIN_CONST_GEN
INC A
ADD A A
INC A
ADD A A
ADD A A
ADD A A
INC A
ADD A A
ADD A A
STORE F # STORE_VAL
SUB A A # BEGIN_CONST_GEN
INC A
INC A
LOAD H
PUT H
HALT