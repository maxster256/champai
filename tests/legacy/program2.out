SUB A A
INC A
INC A
INC A
SUB H H
INC H
INC H
INC H
INC H
INC H
INC H
INC H
STORE H
SUB B B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
INC B
SUB A A
INC A
INC A
INC A
LOAD C
COPY H C
SUB D D
SUB E E
INC E
COPY F B
JZERO C 111
COPY H C
INC H
SUB H B
JZERO H 90
JUMP 93
ADD C C
ADD E E
JUMP 85
COPY H C
SUB H F
JZERO H 97
JUMP 99
SUB F C
ADD D E
HALF C
HALF E
JZERO E 112
COPY H C
SUB H F
JZERO H 106
JUMP 108
SUB F C
ADD D E
HALF C
HALF E
JUMP 101
SUB F F
SUB A A
INC A
INC A
STORE F
SUB A A
INC A
INC A
INC A
LOAD B
COPY H B
SUB A A
INC A
INC A
LOAD C
COPY H C
SUB D D
SUB E E
INC E
COPY F B
JZERO C 158
COPY H C
INC H
SUB H B
JZERO H 137
JUMP 140
ADD C C
ADD E E
JUMP 132
COPY H C
SUB H F
JZERO H 144
JUMP 146
SUB F C
ADD D E
HALF C
HALF E
JZERO E 159
COPY H C
SUB H F
JZERO H 153
JUMP 155
SUB F C
ADD D E
HALF C
HALF E
JUMP 148
SUB F F
SUB A A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
INC A
STORE F
SUB A A
INC A
INC A
LOAD H
PUT H
HALT